src/System.Net.Mail/tests/Functional/SmtpClientTest.cs

```diff
@@ -332,6 +332,82 @@ public async Task TestMailDeliveryAsync()
             }
         }
 
+        [Fact]
+        public void TestSmtpDeliveryFormatSevenBit() {
+            SmtpServer server = null;
+            SmtpClient client = null;
+            var serverList = new System.Collections.Generic.List<SmtpServer>();
+
+            MailMessage msg = new MailMessage("foo@example.com", "bar@example.com");
+            msg.HeadersEncoding = msg.BodyEncoding = msg.SubjectEncoding = System.Text.Encoding.UTF8;
+
+            var subjectTxt = "Test 测试 Contain 包含 UTF8";
+            var subjectEncode = "=?utf-8?B?VGVzdCDmtYvor5UgQ29udGFpbiDljIXlkKsgVVRGOA==?=";
+
+            msg.Subject = subjectTxt;
+
+            Action<SmtpDeliveryFormat, bool, string, string, string> test = (format, utf8, async, sync, tag) => {
+                var retry = 0;
+                while (retry < 3) {
+                    if (client != null) {
+                        server.Stop();
+                        client.Dispose();
+                    }
+                    //New connections are required when parameters change
+                    //How to close cached SmtpConnection !!?? Killing this server is the quickest solution.
+                    server = new SmtpServer();
+                    var running = false;
+                    Thread t = new Thread(() => {
```

Comment:
A new thread should not be used.  If you need to start a server using localhost in the background, please use Tasks.  See other tests in the CoreFx System.Net.Http etc.  for examples.